ARCs/arc-0014.md

```diff
@@ -0,0 +1,67 @@
+---
+ams: 14
+title: Algorand standard for Authentication
+status: Draft
+---
+
+# Algorand Standard for authentication
+
+## Summary
+
+This document introduces the standard for authentication using the algorand accounts.
+
+## Abstract
+
+The goal of this standard is to define authorization process and authentication procedures for communication between web application and backend services or between two backend services. 
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Base64 is the standard described in [RFC-4648](https://www.ietf.org/rfc/rfc4648.txt).
+
+## Request for comments
+
+Please comment or suggest pull request for this document here: [https://github.com/scholtz/AMS/issues](https://github.com/scholtz/AMS/issues)
+
+### Authentication
+
+Authentication is process in which person generates the authorization token for communication with backend service.
+
+Token MUST be generated from Algorand public and private key - from algorand address.
+
+In the first step of the authentication, the message to be signed is generated. The message MUST contain note field with configurable Realm. Realm is identifier of the service. It MAY be the web address of the service for intended purpose.
+
+The message MUST be self signed - the receiver is the same as the sender.
+
+The message MUST have network parameters filled in. The network genesis hash SHOULD be configured for validation pursposes by backend service. The latest block is the time of expiration of the token. The latest block MUST be provided.
+
+In the second step, the message MUST be signed by the account private key.
+
+The message MUST not be submitted to the network (mainnet, testnet) nor should be published in any other way. It may be used only for communcation to the backend service.
```

Comment:
Maybe we can make it such that the transaction will fail, so this is not an issue.

Set fee to 0?

Reply:
i like the idea of having fee below the acceptance on the network.. perhaps 1 should be also good

Reply:
IMHO there should be some "stronger" violation than a low tx fee - one that survives future changes to the network.

Reply:
also, I would love to see two signatures :) 

The first one without any nonce - so I could use the sig as a "password" for end2end or local storage encryption. 
The second one, with a nonce, for authentication .

Maybe, in my case, I would send a group of two "authentication" transactions in one go.  
One with all fields static (no nonce, fv=lv=0) and one with fv=lv=nonce

Reply:
> IMHO there should be some "stronger" violation than a low tx fee - one that survives future changes to the network.

setting the block heights to something in the past or last round < first round in the transaction should ensure that the transaction _cannot_ be accepted by the chain

Reply:
I think forcing the genesis hash and genesis ID to be some specific values should prevent posting the transaction whatever happen.
In addition fee=0, fv=lv=0 looks like an additional good idea.
And everything else in the note field

Reply:
to know the genesis of the network is crucial to check the expiration of the token
keep in mind that you send this token with each request to the backend, as the jwt works.. so we should keep the size to minimum
i think setting the fee to zero is the nicest solution never to add the tx to the network